File: phase 1/02 - sliding windows/lc1004.py
class Solution:
    def longestOnes(self, nums: List[int], k: int) -> int:
        lp = 0
        rp = 0
        cmax = 0
        kcount = 0
        nlen = len(nums)

        while rp < nlen:
            if nums[rp] == 0:
                if kcount < k:
                    kcount+=1
                else:
                    while nums[lp] !=0:
                        lp+=1
                    lp+=1
            cmax = max(cmax, (rp-lp+1))
            rp+=1
        
        return cmax

File: phase 1/02 - sliding windows/lc1004.js
var longestOnes = function(nums, k) {
    let nlen = nums.length

    let lp = 0
    let rp = 0
    let kCounter = 0
    let max = 0

    while (rp < nlen) {
        if (nums[rp] == 0) {
            if (kCounter < k) {
                kCounter++
            } else {
                // Move the left pointer forward until we consume a zero
                while (nums[lp] != 0) {
                    lp++
                }
                lp++ // Move past the zero
            }
        }
        max = Math.max(max, rp - lp + 1)
        rp++
    }

    return max
};
